src/blindpsbt.cpp

```diff
@@ -220,6 +220,31 @@ BlindProofResult VerifyBlindProofs(const PSBTOutput& o) {
     return BlindProofResult::OK;
 }
 
+BlindProofResult VerifyBlindProofs(const PSBTInput& i) {
+    CTxOut utxo;
+    if (!i.GetUTXO(utxo)) {
+        return BlindProofResult::OK;
+    }
+
+    if (i.m_explicit_value != std::nullopt) {
+        if (i.m_value_proof.empty()) {
+            return BlindProofResult::MISSING_VALUE_PROOF;
+        } else if (!VerifyBlindValueProof(*i.m_explicit_value, utxo.nValue, i.m_value_proof, utxo.nAsset)) {
```

Comment:
In ceee8a6abafe368775b5c71a38d3973691f134bb:

Need to check that `utxo.nValue` is confidential, otherwise we potentially wind up passing uninitialized memory to the libsecp functions. If it's not confidential it's fine to just return an error.

Reply:
Done. Also added the same check below for `nAsset`.